test/unit/org/apache/cassandra/repair/autorepair/RepairTokenRangeSplitterTest.java

```diff
@@ -81,16 +81,19 @@ public void setUp()
     public void testSizePartitionCount() throws Throwable
     {
         insertAndFlushTable(tableName, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
-        Refs<SSTableReader> sstables = RepairTokenRangeSplitter.getSSTableReaderRefs(RepairType.FULL, KEYSPACE, tableName, FULL_RANGE);
-        assertEquals(10, sstables.iterator().next().getEstimatedPartitionSize().count());
-        SizeEstimate sizes = RepairTokenRangeSplitter.getSizesForRangeOfSSTables(RepairType.FULL, KEYSPACE, tableName, FULL_RANGE, sstables);
-        assertEquals(10, sizes.partitions);
+        try (Refs<SSTableReader> sstables = RepairTokenRangeSplitter.getSSTableReaderRefs(RepairType.FULL, KEYSPACE, tableName, FULL_RANGE))
+        {
+            assertEquals(10, sstables.iterator().next().getEstimatedPartitionSize().count());
+            SizeEstimate sizes = RepairTokenRangeSplitter.getSizesForRangeOfSSTables(RepairType.FULL, KEYSPACE, tableName, FULL_RANGE, sstables);
+            assertEquals(10, sizes.partitions);
+        }
     }
 
     @Test
     public void testSizePartitionCountSplit() throws Throwable
     {
-        int[] values = new int[10000];
+        int partitionCount = 100_000;
```

Comment:
Updated test to use a larger sample size,  because we need, the previous test expected a very small margin (.6%) that was observed in testing.  With this sample size, the partition count is overestimated by about 2800 (~2.8%) because the compression blocks cause us to overestimate.  If I turn off compression its very accurate (<100), but i figured it would be appropriate to test with compression by default.

Reply:
Yeah, by default, with compression, it makes sense. Should we create a P1 ticket to have tests with non-compression?

Reply:
I think we're alright to not test for it specifically now that we defer to `onDiskSizeForPartitionPositions` which has a variety of tests factoring in compression in `SSTableReaderTest` and there's also no specific logic in `RepairTokenRangeSplitter`.